khmer/_oxli/graphs.pyx

```diff
@@ -1,20 +1,258 @@
 # cython: c_string_type=unicode, c_string_encoding=utf8
+from math import log
+
 from cython.operator cimport dereference as deref
+from libc.stdint cimport uint64_t
+
+from libcpp.memory cimport unique_ptr
+from libcpp.vector cimport vector
 
-from .._khmer import Countgraph, Nodegraph, GraphLabels 
+from .utils cimport _bstring
+from graphs cimport CpQFCounttable
+from parsing cimport CpFastxReader, CPyReadParser_Object
+from oxli_types cimport MAX_BIGCOUNT
+from .._khmer import Countgraph, Nodegraph, GraphLabels, Nodetable, ReadParser
 
 
 cdef CpHashgraph * get_hashgraph_ptr(object graph):
     if not (isinstance(graph, Countgraph) or isinstance(graph, Nodegraph)):
         return NULL
-    
+
     cdef CPyHashgraph_Object* ptr = <CPyHashgraph_Object*> graph
     return deref(ptr).hashgraph
-    
+
 
 cdef CpLabelHash * get_labelhash_ptr(object labels):
     if not isinstance(labels, GraphLabels):
         return NULL
-    
+
     cdef CPyGraphLabels_Object * ptr = <CPyGraphLabels_Object*> labels
     return deref(ptr).labelhash
+
+
+cdef class QFCounttable:
+    cdef unique_ptr[CpQFCounttable] c_table
+    def __cinit__(self, int k, int starting_size):
+        # starting size has to be a power of two
+        power_of_two = ((starting_size & (starting_size - 1) == 0) and
+                        (starting_size != 0))
+        if not power_of_two:
+            raise ValueError("starting_size has to be a power of two.")
+        self.c_table.reset(new CpQFCounttable(k, int(log(starting_size, 2))))
+
+    def add(self, kmer):
+        """Increment the count of this k-mer.
+
+        Synonym for 'count'.
+        """
+        return self.count(kmer)
+
+    def count(self, kmer):
+        """Increment the count of this k-mer
+
+        `kmer` can be either a string or an integer representing the hashed
+        value of the kmer.
+        """
+        if isinstance(kmer, str):
+            temp = kmer.encode('utf-8')
```

Comment:
You can't define a `cdef` inside the if statement, so you could declare it outside and then assign here but I'd claim that is even more clunky than the cast.